### DISCUSSION TITLE
How does colocated git know to ignore `.jj/`?

### DISCUSSION THREAD

**Discussion by miquella** [jj-vcs/jj#5185] [General]
I've recently been working with a tool (Prettier) that scans through all files in the working copy, automatically ignoring everything in `.gitignore`. However, I noticed that it's also scanning through files in the `.jj/` directory as well, which isn't helpful and is making it take longer than it should.

I checked `git status --ignored` and it clearly lists `.jj/` as ignored, but I can't for the life of me figure out how `git` knows to ignore it!

```console
$ git status --ignored
HEAD detached at 815ae96
Ignored files:
  (use "git add -f <file>..." to include in what will be committed)
        .jj/
```

Does anyone happen to know what I might be missing? How does `git` know to ignore `.jj/`?

**Comment by bnjmnt4n** [comment]
The `.jj` directory has a `.gitignore` file with contents `/*`, causing the whole directory to be ignored.

(It does seem like `.jj` has been added to Prettier's native ignore list: https://github.com/prettier/prettier/pull/16684, so updating Prettier might also fix your issue).

**Reply by miquella** [reply]
Ah, thank you! That was what I was missing, so no more losing sleep over that tonight 😅

(I'll go push that upgrade through right now)
